tmt/steps/provision/mrack.py

```diff
@@ -0,0 +1,596 @@
+import dataclasses
+import datetime
+import logging
+import os
+import sys
+from typing import Any, Dict, List, Optional, Tuple, cast
+
+import click
+
+import tmt
+import tmt.options
+import tmt.steps
+import tmt.steps.provision
+import tmt.utils
+from tmt.utils import ProvisionError, updatable_message
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+import asyncio
+from functools import wraps
+
+mrack = Any
+providers = Any
+ProvisioningError = Any
+NotAuthenticatedError = Any
+BEAKER = Any
+BeakerProvider = Any
+BeakerTransformer = Any
+TmtBeakerTransformer = Any
+
+DEFAULT_USER = 'root'
+DEFAULT_ARCH = 'x86_64'
+DEFAULT_PROVISION_TIMEOUT = 3600  # 1 hour timeout at least
+DEFAULT_PROVISION_TICK = 60  # poll job each minute
+
+
+# Type annotation for "data" package describing a guest instance. Passed
+# between load() and save() calls
+GuestInspectType = TypedDict(
+    'GuestInspectType', {
+        "status": str,
+        "system": str,
+        'address': Optional[str]
+        }
+    )
+
+size_translation = {
+    "TB": 1000000,
+    "GB": 1000,
+    "MB": 1,
+    "TiB": 1048576,
+    "GiB": 1024,
+    "MiB": 1,
+    }
+
+operators = [
+    "~=",
+    ">=",
+    "<=",
+    ">",
+    "<",
+    "=",
+    ]
+
+
+def import_and_load_mrack_deps() -> None:
+    """
+    Import mrack module only when needed
+
+    Until we have a separate package for each plugin.
+    """
+    global mrack
+    global providers
+    global ProvisioningError
+    global NotAuthenticatedError
+    global BEAKER
+    global BeakerProvider
+    global BeakerTransformer
+    global TmtBeakerTransformer
+
+    try:
+        import mrack
+        from mrack.errors import NotAuthenticatedError, ProvisioningError
+        from mrack.providers import providers
+        from mrack.providers.beaker import PROVISIONER_KEY as BEAKER
+        from mrack.providers.beaker import BeakerProvider
+        from mrack.transformers.beaker import BeakerTransformer
+
+        # HAX remove mrack stdout and move the logfile to /tmp
+        mrack.logger.removeHandler(mrack.console_handler)
+        mrack.logger.removeHandler(mrack.file_handler)
+        if os.stat("mrack.log"):
+            os.remove("mrack.log")
+        mrack.logger.addHandler(logging.FileHandler("/tmp/mrack.log"))
```

Comment:
> Sorry for nitpicking but tmt has the concepts of workdir - could the logger be in provision step workdir?

Maybe even include the guest name, or is it included automatically in the path?

```python
logging.FileHandler(str(self.workdir / f'{self.name}-mrack.log'))
```

> Or could we somehow move the logger to the tmt's ? Is it possible use tmt's handler in mrack, @happz ?

Good point, but tmt's logging has not been tested when exposing its logger to other libraries - I definitely plan to do something like that for `fmf`, so it will be solved, but ready yet, and I'd rather not block this PR on it. The swapping of handlers should work for now.

Reply:
Sure I do not like this hax as well so let me update it to @happz suggestion, Thanks.